app/renderer/components/bookmarks/bookmarkToolbarButton.js

```diff
@@ -146,30 +135,40 @@ class BookmarkToolbarButton extends React.Component {
   }
 
   openContextMenu (e) {
-    contextMenus.onSiteDetailContextMenu(this.props.bookmark, this.activeFrame, e)
+    if (e) {
+      e.stopPropagation()
+    }
+    windowActions.onSiteDetailMenu(this.props.bookmarkKey)
   }
 
   clickBookmarkItem (e) {
-    return bookmarkActions.clickBookmarkItem(this.props.bookmarks, this.props.bookmark, this.activeFrame, e)
+    return bookmarkActions.clickBookmarkItem(this.props.bookmarkKey, this.props.tabId, e)
   }
 
   showBookmarkFolderMenu (e) {
+    const rectLeft = e.target.getBoundingClientRect()
+    const rectBottom = e.target.parentNode.getBoundingClientRect()
+    const left = (rectLeft.left | 0) - 2
+    const top = (rectBottom.bottom | 0) - 1
+
+    if (e && e.stopPropagation) {
+      e.stopPropagation()
+    }
+
+    windowActions.onShowBookmarkFolderMenu(this.props.bookmarkKey, left, top)
```

Comment:
can you add a TODO to merge `setBookmarksToolbarSelectedFolderId` into `onShowBookmarkFolderMenu`. We should never call more than one action at a time